core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java

```diff
@@ -378,31 +379,43 @@ public NativeImageSourceJarBuildItem buildNativeImageJar(CurateOutcomeBuildItem
      */
     private void copyJsonConfigFiles(ApplicationArchivesBuildItem applicationArchivesBuildItem, Path thinJarDirectory)
             throws IOException {
-        // archiveLocation contains the location of the application jar at this point
-        Path appJarPath = applicationArchivesBuildItem.getRootArchive().getArchiveLocation();
-        try (FileSystem jarFileSystem = FileSystems.newFileSystem(appJarPath, null)) {
-            try (Stream<Path> stream = Files.find(jarFileSystem.getPath("/"), 1, new BiPredicate<Path, BasicFileAttributes>() {
+        Path archiveLocation = applicationArchivesBuildItem.getRootArchive().getArchiveLocation();
+        if (Files.isDirectory(archiveLocation)) { // this is the use case where the native image is built as part of the regular build
+            Files.find(archiveLocation, 1, IS_JSON_FILE_PREDICATE).forEach(new Consumer<Path>() {
```

Comment:
Files.find needs to be in a try with resources block, as the stream needs to be closed.

Reply:
Oops! My bad, it should be fixed now